holmes/plugins/toolsets/infrainsights/redis_toolset.py

```
@@ -0,0 +1,534 @@
+                ssl_cert_reqs=ssl_cert_reqs,
+                decode_responses=True,
+                socket_timeout=5,
+                socket_connect_timeout=5
+            )
+            
+            # Test connection
+            self.client.ping()
+            
+        except Exception as e:
+            logging.error(f"Failed to connect to Redis: {e}")
+            raise Exception(f"Redis connection failed: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)